pkg/docker/image_resolver.go

```diff
@@ -0,0 +1,42 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config"
+	"github.com/rs/zerolog/log"
+)
+
+type imageResolverFunc func(context.Context, string, config.DockerCredentials) (string, error)
+
+// ResolveImageID will take the provided image identifier and a resolver,
+// and attempt to provide a version of the image id containing the digest
+// instead.
+func ResolveImageID(ctx context.Context, i ImageID, resolver imageResolverFunc) (ImageID, error) {
+	if i.HasDigest() {
+		return i, nil
+	}
+
+	// TODO: Look up i.String() in cache to see if we already have a digest for it
```

Comment:
We'll need to do this carefully, right? If the `ImageID` contains the `latest` tag we'll want to avoid caching it, correct? (Mostly asking to clarify my understanding of the code)

Reply:
It's really just a marker for me to remind me where to do the lookup when I get to https://github.com/bacalhau-project/bacalhau/issues/2254  - this suggests caching the digest/manifest for ~1 hour for latest/no-tag but that might affect users who are in a dev/push/test cycle with untagged images, so may need more thought.